### PR TITLE
[dynamo] Add `itertools.repeat` via polyfill

### LINKED ISSUE #110286: [dynamo] support itertools.repeat: call_function repeat in skip_files Builtin repeat, skip reason: filename is None
### 🐛 Describe the bug

self explanatory

we should use the polyfill approach to implement this see also https://github.com/pytorch/pytorch/pull/109284#issuecomment-1737533266 https://github.com/pytorch/pytorch/pull/109972#discussion_r1336174289

Real world use case in torchrec dist_data.py

```
            batch_size_per_feature = list(
                itertools.chain.from_iterable(
                    itertools.repeat(x, local_split) for x in batch_size_per_rank
                )
            )
```

### Versions

main

cc @msaroufim @wconstab @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### PR DESCRIPTION
**Pull request by jon-chuang**
Fixes https://github.com/pytorch/pytorch/issues/110286

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @ezyang

### CODE REVIEW

File: torch/_dynamo/variables/misc.py [hunk 1]
```diff
@@ -914,6 +914,17 @@ def wraps(fn):
             return variables.functions.FunctoolsPartialVariable(
                 fn, args=rest_args, keywords=kwargs, **options
             )
+        elif self.value is itertools.repeat:
+            from .builder import SourcelessBuilder
+
+            if len(args) < 2:
+                # We cannot risk infinite generator being consumed to exhaustion by dynamo
+                # (i.e. infinite loop)
```
Comment (ezyang) [inline torch/_dynamo/variables/misc.py:922]: How come? Shouldn't this be triggering the generator protocol?

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/misc.py (hunk 1 above)
Comment (jon-chuang) [reply torch/_dynamo/variables/misc.py:922]: Hmm, for some reason, it hangs.

I implemented it as:
```python
if count is None:
  while True:
    yield item
```

Comment (jon-chuang) [reply torch/_dynamo/variables/misc.py:922]: The reason is that this does not get wrapped into a generator, but rather a `ListIteratorVariable` for some reason.

Comment (jon-chuang) [reply torch/_dynamo/variables/misc.py:922]: In particular, InliningGeneratorTranslator simply yields the entire generator into a list, which is then returned as a `ListIteratorVariable`:
https://github.com/pytorch/pytorch/blob/8bc04f46fe8e69188fa46f1611b46788a7d4824d/torch/_dynamo/symbolic_convert.py#L2518

**Pull request closed without merging**